sources/org.osbuild.curl

```diff
@@ -86,6 +87,59 @@ SCHEMA = """
 """
 
 
+class Subscriptions:
+    def __init__(self, sections):
+        self.sections = sections
+
+    @classmethod
+    def from_host_system(cls):
+        with open('/etc/yum.repos.d/redhat.repo', 'r') as fp:
+            return cls._parse_repo_file(fp)
+
+    @staticmethod
+    def _process_baseurl(input_url):
+        # First escape meta characters: . ^ $ * + ? { } [ ] \ | ( )
+        # Except for $basearch, which will be replaced later.
+        for character in ['\\', '.', '^', '*', '+', '?', '{', '}', '[', ']', '|', '(',')']:
+            input_url = input_url.replace(character, f"\\{character}")
```

Comment:
Did you consider using `re.escape(pattern)` method instead of implementing it yourself? Sure, the `$` would be escaped as well, but that does not make it impossible to replace necessary variables in the next step, you'll just need to change the string that is being replaced.

Reply:
I didn't know about this function, thanks :+1: